presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java

```diff
@@ -191,6 +191,9 @@ public CacheStatsMBean getFilterCache()
         try {
             pageProjectionWorkClass = defineClass(pageProjectionWorkDefinition, Work.class, callSiteBinder.getBindings(), getClass().getClassLoader());
         }
+        catch (PrestoException prestoException) {
```

Comment:
What's the point of this code? I think you want to catch `ByteCodeTooLargeException`?

Reply:
Oh i see you did that in `CompilerUtils` so this should not be needed?

Reply:
So if I don't have it here, it's not getting surfaced as user error. It will become COMPILER_ERROR which we don't want

Reply:
what? why? that doesn't make sense? Why catching and rethrow it would change the error type?